doc/tutorials/enable_s5.rst

```diff
@@ -0,0 +1,367 @@
+.. _enable-s5:
+
+Platform S5 Enable Guide
+########################
+
+Introduction
+************
+
+S5 is one of the `ACPI Sleep State <http://acpi.sourceforge.net/documentation/sleep.html>`_.
+It refers to the system being shut down (although some power may still be supplied to
+certain devices). In this document, S5 means the function to shut down the
+**User VMs**, **the Service VM**,  the hypervisor, and the hardware. In most cases,
+directly shutting down the power of a computer system is not advisable because it can
+damage some components. It can cause corruption and put the system in an unknown or
+unstable state. On ACRN, the User VM must be shut down before powering off the Service VM.
+Especially for some use cases, where User VMs could be used in industrial control or other
+high safety requirement environment, a graceful system shutdown such as the ACRN S5
+function is required.
+
+S5 Architecture
+***************
+
+ACRN provides a mechanism to trigger the S5 state transition throughout the system.
+It uses vUART channel to communicate between the Service VM and the User VMs.
+The diagram below shows the overall architecture:
+
+.. figure:: images/s5_overall_architecture.png
+   :align: center
+
+   S5 overall architecture
+
+- **Scenario I**:
+
+    The User VM's serial port device (``ttySn``) is emulated in Device Model, the channel from Service VM to User VM:
+
+    .. graphviz:: images/s5-scenario-1.dot
+       :name: s5-scenario-1
+
+- **Scenario II**:
+
+    The User VM's (like RT-Linux or other RT-VMs) serial port device (``ttySn``) is emulated in Hypervisor,
+    the channel from Service OS to User VM:
+
+    .. graphviz:: images/s5-scenario-2.dot
+       :name: s5-scenario-2
+
+How to trigger User VM's S5?
+============================
+
+On Service VM side, it uses the ``acrnctl`` tool to trigger the User VM's S5 flow:
+``acrnctl stop user-vm-name``. Then Device Model will send a ``shutdown`` command
+to User VM through channel, if User VM receives the command, it will send an "ACK"
+to Device Model. It is the responsibility of Service VM to check if the User VMs
+shutdown successfully or not, and decides when to power off itself.
+
+What does "life cycle manager" in User VM do?
+=============================================
+
+As current S5 reference design, there is a life-cycle-manager daemon running in
+User VM to implement S5. It will wait for the command from Service VM on the
+paired serial port, there is a simple protocol between Service VM and User VM:
+when it receives ``shutdown``, then send "acked" to Service VM, then it can power off
+User VM.
+
+However, if the User VM is not ready to power off, it can ignore the ``shutdown`` command.
+
+.. _enable_s5:
+
+Enable S5
+*********
+
+Enable S5 based on the the particular OS:
+
+* For Linux (LaaG) or Windows (WaaG), refer to the following configurations in
+  ``devicemodel/samples/nuc/launch_uos.sh`` launch script for ``acrn-dm``.
+
+  .. code-block:: none
```

Comment:
I'm a little concerned that this is a copy of some lines from the launch_uos.sh script that could get out of sync if that script changes.  Better would be to do an include of that script specifying which lines to include here.